assets/js/base/context/cart-checkout/payment-methods/payment-method-data-context.js

```diff
@@ -103,26 +104,24 @@ const isErrorResponse = ( response ) => {
  * @param {Object} props                     Incoming props for provider
  * @param {Object} props.children            The wrapped components in this
  *                                           provider.
- * @param {string} props.activePaymentMethod The initial active payment method
- *                                           to set for the context.
  */
-export const PaymentMethodDataProvider = ( {
-	children,
-	activePaymentMethod: initialActivePaymentMethod,
-} ) => {
+export const PaymentMethodDataProvider = ( { children } ) => {
 	const { setBillingData } = useBillingDataContext();
 	const {
 		isComplete: checkoutIsComplete,
 		isProcessingComplete: checkoutIsProcessingComplete,
 		isCalculating: checkoutIsCalculating,
 		hasError: checkoutHasError,
 	} = useCheckoutContext();
-	const [ activePaymentMethod, setActive ] = useState(
-		initialActivePaymentMethod
-	);
```

Comment:
it seems even when you fix the issues I lined in other comments, the payment data is still lacking some data, as `<Tabs>` will fail because you're not setting `initialActivePaymentMethod` to anything before passing it (it will default to {}), if this is an expected behavior (to not set it in the editor) you should probably change the check in tabs

https://github.com/woocommerce/woocommerce-gutenberg-products-block//blob/ee73458babfb6ef8d35987491d2f5900280f6255/assets/js/base/components/tabs/index.js#L26-L28

because `{}` is a truthy value, so `useState` will use that value, a better check would be this
```js
const [ selected, setSelected ] = useState(
		Object.keys( initialTabName ).length || ( tabs.length > 0 ? tabs[ 0 ].name : '' )
);
```

Reply:
No the issue is that the initial state was changed to `{}` instead of `''` (see my comment below)

Reply:
yes, that's what I noted for the reason this is breaking, `''` is falsy while `{}` is truthy.

Reply:
Right, sorry, to clarify I'm suggesting that the fix is simply to change the default (not change the tab logic).